gax/src/main/java/com/google/api/gax/rpc/StubSettings.java

```diff
@@ -177,11 +189,29 @@ protected Builder(StubSettings settings) {
       this.internalHeaderProvider = settings.internalHeaderProvider;
       this.clock = settings.clock;
       this.endpoint = settings.endpoint;
+      this.quotaProjectID = settings.quotaProjectID;
       this.streamWatchdogProvider = settings.streamWatchdogProvider;
       this.streamWatchdogCheckInterval = settings.streamWatchdogCheckInterval;
       this.tracerFactory = settings.tracerFactory;
     }
 
+    /** Get Quota Project ID from Client Context * */
+    private static String getQuotaProjectIDFromClientContext(ClientContext clientContext) {
```

Comment:
Seems like we use `ID` in methods and variables, and `Id` in class names. Is this idiomatic?

Reply:
`ID` used in all methods and variables, which is really just my preference. I haven't find any guideline for naming.
Let me know if any rule need to be follow here, I could make change.

Reply:
We should use `quotaProjectId`. See https://google.github.io/styleguide/javaguide.html#s5-naming

Reply:
I will make change to `quotaProjectId`